File: .sdk-repos/orchestration-cluster-api-js/examples/additional-operations.ts
// Compilable usage examples for decision instances, requirements, documents, and other operations.
// These examples are type-checked during build to guard against API regressions.

import {
  createCamundaClient,
  type DecisionDefinitionKey,
  type DecisionEvaluationInstanceKey,
  type DecisionEvaluationKey,
  type DecisionRequirementsKey,
  type DocumentId,
  type JobKey,
  type UserTaskKey,
} from '@camunda8/orchestration-cluster-api';

//#region GetDecisionDefinitionXml
async function getDecisionDefinitionXmlExample(decisionDefinitionKey: DecisionDefinitionKey) {
  const camunda = createCamundaClient();

  const xml = await camunda.getDecisionDefinitionXml(
    { decisionDefinitionKey },
    { consistency: { waitUpToMs: 5000 } }
  );

  console.log(`XML length: ${JSON.stringify(xml).length}`);
}
//#endregion GetDecisionDefinitionXml

//#region GetDecisionInstance
async function getDecisionInstanceExample(
  decisionEvaluationInstanceKey: DecisionEvaluationInstanceKey
) {
  const camunda = createCamundaClient();

  const instance = await camunda.getDecisionInstance(
    { decisionEvaluationInstanceKey },
    { consistency: { waitUpToMs: 5000 } }
  );

  console.log(`Decision: ${instance.decisionDefinitionId}`);
}
//#endregion GetDecisionInstance

//#region SearchDecisionInstances
async function searchDecisionInstancesExample() {
  const camunda = createCamundaClient();

  const result = await camunda.searchDecisionInstances(
    {
      page: { limit: 10 },
    },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const instance of result.items ?? []) {
    console.log(`${instance.decisionEvaluationKey}: ${instance.decisionDefinitionId}`);
  }
}
//#endregion SearchDecisionInstances

//#region DeleteDecisionInstance
async function deleteDecisionInstanceExample(decisionEvaluationKey: DecisionEvaluationKey) {
  const camunda = createCamundaClient();

  await camunda.deleteDecisionInstance({ decisionEvaluationKey });
}
//#endregion DeleteDecisionInstance

//#region GetDecisionRequirements
async function getDecisionRequirementsExample(decisionRequirementsKey: DecisionRequirementsKey) {
  const camunda = createCamundaClient();

  const requirements = await camunda.getDecisionRequirements(
    { decisionRequirementsKey },
    { consistency: { waitUpToMs: 5000 } }
  );

  console.log(`Requirements: ${requirements.decisionRequirementsId}`);
}
//#endregion GetDecisionRequirements

//#region GetDecisionRequirementsXml
async function getDecisionRequirementsXmlExample(decisionRequirementsKey: DecisionRequirementsKey) {
  const camunda = createCamundaClient();

  const xml = await camunda.getDecisionRequirementsXml(
    { decisionRequirementsKey },
    { consistency: { waitUpToMs: 5000 } }
  );

  console.log(`XML length: ${JSON.stringify(xml).length}`);
}
//#endregion GetDecisionRequirementsXml

//#region SearchDecisionRequirements
async function searchDecisionRequirementsExample() {
  const camunda = createCamundaClient();

  const result = await camunda.searchDecisionRequirements(
    {
      page: { limit: 10 },
    },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const req of result.items ?? []) {
    console.log(`${req.decisionRequirementsKey}: ${req.decisionRequirementsId}`);
  }
}
//#endregion SearchDecisionRequirements

//#region CreateDocumentLink
async function createDocumentLinkExample(documentId: DocumentId) {
  const camunda = createCamundaClient();

  const link = await camunda.createDocumentLink({
    documentId,
    timeToLive: 3600000,
  });

  console.log(`Document link: ${link.url}`);
}
//#endregion CreateDocumentLink

//#region DeleteDocument
async function deleteDocumentExample(documentId: DocumentId) {
  const camunda = createCamundaClient();

  await camunda.deleteDocument({ documentId });
}
//#endregion DeleteDocument

//#region ThrowJobError
async function throwJobErrorExample(jobKey: JobKey) {
  const camunda = createCamundaClient();

  await camunda.throwJobError({
    jobKey,
    errorCode: 'PAYMENT_FAILED',
    errorMessage: 'Payment provider returned error',
  });
}
//#endregion ThrowJobError

//#region UpdateJob
async function updateJobExample(jobKey: JobKey) {
  const camunda = createCamundaClient();

  await camunda.updateJob({
    jobKey,
    changeset: { retries: 5, timeout: 60000 },
  });
}
//#endregion UpdateJob

//#region SearchJobs
async function searchJobsExample() {
  const camunda = createCamundaClient();

  const result = await camunda.searchJobs(
    {
      filter: { type: 'payment-processing', state: 'CREATED' },
      page: { limit: 10 },
    },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const job of result.items ?? []) {
    console.log(`Job ${job.jobKey}: ${job.type} (${job.state})`);
  }
}
//#endregion SearchJobs

//#region GetUserTask
async function getUserTaskExample(userTaskKey: UserTaskKey) {
  const camunda = createCamundaClient();

  const task = await camunda.getUserTask({ userTaskKey }, { consistency: { waitUpToMs: 5000 } });

  console.log(`Task: ${task.name} (${task.state})`);
}
//#endregion GetUserTask

//#region UpdateUserTask
async function updateUserTaskExample(userTaskKey: UserTaskKey) {
  const camunda = createCamundaClient();

  await camunda.updateUserTask({
    userTaskKey,
    changeset: {
      candidateUsers: ['alice', 'bob'],
      dueDate: '2025-12-31T23:59:59Z',
      priority: 80,
    },
  });
}
//#endregion UpdateUserTask

//#region GetUserTaskForm
async function getUserTaskFormExample(userTaskKey: UserTaskKey) {
  const camunda = createCamundaClient();

  const form = await camunda.getUserTaskForm(
    { userTaskKey },
    { consistency: { waitUpToMs: 5000 } }
  );

  if (form) {
    console.log(`Form key: ${form.formKey}`);
  }
}
//#endregion GetUserTaskForm

//#region SearchUserTaskVariables
async function searchUserTaskVariablesExample(userTaskKey: UserTaskKey) {
  const camunda = createCamundaClient();

  const result = await camunda.searchUserTaskVariables(
    { userTaskKey },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const variable of result.items ?? []) {
    console.log(`${variable.name} = ${variable.value}`);
  }
}
//#endregion SearchUserTaskVariables

//#region SearchUserTaskAuditLogs
async function searchUserTaskAuditLogsExample(userTaskKey: UserTaskKey) {
  const camunda = createCamundaClient();

  const result = await camunda.searchUserTaskAuditLogs(
    { userTaskKey },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const log of result.items ?? []) {
    console.log(`Audit: ${log.operationType} at ${log.timestamp}`);
  }
}
//#endregion SearchUserTaskAuditLogs

//#region CreateDocument
async function createDocumentExample() {
  const camunda = createCamundaClient();

  const file = new Blob(['Hello, world!'], { type: 'text/plain' });

  const result = await camunda.createDocument({
    file,
    metadata: { fileName: 'hello.txt' },
  });

  console.log(`Document ID: ${result.documentId}`);
}
//#endregion CreateDocument

//#region CreateDocuments
async function createDocumentsExample() {
  const camunda = createCamundaClient();

  const file1 = new Blob(['File one'], { type: 'text/plain' });
  const file2 = new Blob(['File two'], { type: 'text/plain' });

  const result = await camunda.createDocuments({
    files: [file1, file2],
    metadataList: [{ fileName: 'one.txt' }, { fileName: 'two.txt' }],
  });

  for (const doc of result.createdDocuments ?? []) {
    console.log(`Created: ${doc.documentId}`);
  }
}
//#endregion CreateDocuments

//#region GetDocument
async function getDocumentExample(documentId: DocumentId) {
  const camunda = createCamundaClient();

  await camunda.getDocument({ documentId });

  console.log(`Downloaded document: ${documentId}`);
}
//#endregion GetDocument

//#region SearchUserTaskEffectiveVariables
async function searchUserTaskEffectiveVariablesExample(userTaskKey: UserTaskKey) {
  const camunda = createCamundaClient();

  const result = await camunda.searchUserTaskEffectiveVariables(
    { userTaskKey },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const variable of result.items ?? []) {
    console.log(`${variable.name} = ${variable.value}`);
  }
}
//#endregion SearchUserTaskEffectiveVariables

// Suppress "declared but never read"
void getDecisionDefinitionXmlExample;
void getDecisionInstanceExample;
void searchDecisionInstancesExample;
void deleteDecisionInstanceExample;
void getDecisionRequirementsExample;
void getDecisionRequirementsXmlExample;
void searchDecisionRequirementsExample;
void createDocumentLinkExample;
void deleteDocumentExample;
void throwJobErrorExample;
void updateJobExample;
void searchJobsExample;
void getUserTaskExample;
void updateUserTaskExample;
void getUserTaskFormExample;
void searchUserTaskVariablesExample;
void searchUserTaskAuditLogsExample;
void createDocumentExample;
void createDocumentsExample;
void getDocumentExample;
void searchUserTaskEffectiveVariablesExample;

File: .sdk-repos/orchestration-cluster-api-python/examples/additional_operations.py
# Compilable usage examples for decision instances, documents, jobs, and user task extras.
# These examples are type-checked during build to guard against API regressions.
from __future__ import annotations

from camunda_orchestration_sdk import (
    CamundaClient,
    CreateDocumentData,
    CreateDocumentsData,
    DecisionDefinitionKey,
    DecisionEvaluationInstanceKey,
    DecisionEvaluationKey,
    DecisionInstanceSearchQuery,
    DecisionRequirementsKey,
    DecisionRequirementsSearchQuery,
    DocumentId,
    DocumentLinkRequest,
    File,
    JobChangeset,
    JobErrorRequest,
    JobKey,
    JobSearchQuery,
    JobUpdateRequest,
    Unset,
    UserTaskAuditLogSearchQueryRequest,
    UserTaskKey,
)


# region GetDecisionDefinitionXml
def get_decision_definition_xml_example(decision_definition_key: DecisionDefinitionKey) -> None:
    client = CamundaClient()

    xml = client.get_decision_definition_xml(
        decision_definition_key=decision_definition_key,
    )

    print(f"XML length: {len(xml)}")
# endregion GetDecisionDefinitionXml


# region GetDecisionInstance
def get_decision_instance_example(decision_evaluation_instance_key: DecisionEvaluationInstanceKey) -> None:
    client = CamundaClient()

    result = client.get_decision_instance(
        decision_evaluation_instance_key=decision_evaluation_instance_key,
    )

    print(f"Decision instance: {result.decision_definition_id}")
# endregion GetDecisionInstance


# region SearchDecisionInstances
def search_decision_instances_example() -> None:
    client = CamundaClient()

    result = client.search_decision_instances(
        data=DecisionInstanceSearchQuery(),
    )

    if not isinstance(result.items, Unset):
        for di in result.items:
            print(f"Decision instance: {di.decision_definition_id}")
# endregion SearchDecisionInstances


# region DeleteDecisionInstance
def delete_decision_instance_example(decision_evaluation_key: DecisionEvaluationKey) -> None:
    client = CamundaClient()

    client.delete_decision_instance(
        decision_evaluation_key=decision_evaluation_key,
    )
# endregion DeleteDecisionInstance


# region GetDecisionRequirements
def get_decision_requirements_example(decision_requirements_key: DecisionRequirementsKey) -> None:
    client = CamundaClient()

    result = client.get_decision_requirements(
        decision_requirements_key=decision_requirements_key,
    )

    print(f"DRD: {result.decision_requirements_name}")
# endregion GetDecisionRequirements


# region GetDecisionRequirementsXml
def get_decision_requirements_xml_example(decision_requirements_key: DecisionRequirementsKey) -> None:
    client = CamundaClient()

    xml = client.get_decision_requirements_xml(
        decision_requirements_key=decision_requirements_key,
    )

    print(f"XML length: {len(xml)}")
# endregion GetDecisionRequirementsXml


# region SearchDecisionRequirements
def search_decision_requirements_example() -> None:
    client = CamundaClient()

    result = client.search_decision_requirements(
        data=DecisionRequirementsSearchQuery(),
    )

    if not isinstance(result.items, Unset):
        for drd in result.items:
            print(f"DRD: {drd.decision_requirements_name}")
# endregion SearchDecisionRequirements


# region CreateDocumentLink
def create_document_link_example(document_id: DocumentId) -> None:
    client = CamundaClient()

    result = client.create_document_link(
        document_id=document_id,
        data=DocumentLinkRequest(),
    )

    print(f"Document link: {result.url}")
# endregion CreateDocumentLink


# region DeleteDocument
def delete_document_example(document_id: DocumentId) -> None:
    client = CamundaClient()

    client.delete_document(document_id=document_id)
# endregion DeleteDocument


# region ThrowJobError
def throw_job_error_example(job_key: JobKey) -> None:
    client = CamundaClient()

    client.throw_job_error(
        job_key=job_key,
        data=JobErrorRequest(
            error_code="VALIDATION_ERROR",
            error_message="Input validation failed",
        ),
    )
# endregion ThrowJobError


# region UpdateJob
def update_job_example(job_key: JobKey) -> None:
    client = CamundaClient()

    client.update_job(
        job_key=job_key,
        data=JobUpdateRequest(
            changeset=JobChangeset(
                retries=3,
            ),
        ),
    )
# endregion UpdateJob


# region SearchJobs
def search_jobs_example() -> None:
    client = CamundaClient()

    result = client.search_jobs(
        data=JobSearchQuery(),
    )

    if not isinstance(result.items, Unset):
        for job in result.items:
            print(f"Job: {job.job_key}")
# endregion SearchJobs


# region GetUserTask
def get_user_task_example(user_task_key: UserTaskKey) -> None:
    client = CamundaClient()

    result = client.get_user_task(
        user_task_key=user_task_key,
    )

    print(f"User task: {result.user_task_key}")
# endregion GetUserTask


# region UpdateUserTask
def update_user_task_example(user_task_key: UserTaskKey) -> None:
    from camunda_orchestration_sdk import Changeset, UserTaskUpdateRequest

    client = CamundaClient()

    client.update_user_task(
        user_task_key=user_task_key,
        data=UserTaskUpdateRequest(
            changeset=Changeset(
                priority=80,
            ),
        ),
    )
# endregion UpdateUserTask


# region GetUserTaskForm
def get_user_task_form_example(user_task_key: UserTaskKey) -> None:
    client = CamundaClient()

    result = client.get_user_task_form(
        user_task_key=user_task_key,
    )

    print(f"Form: {result.form_key}")
# endregion GetUserTaskForm


# region SearchUserTaskVariables
def search_user_task_variables_example(user_task_key: UserTaskKey) -> None:
    client = CamundaClient()

    result = client.search_user_task_variables(
        user_task_key=user_task_key,
    )

    if not isinstance(result.items, Unset):
        for var in result.items:
            print(f"Variable: {var.name}")
# endregion SearchUserTaskVariables


# region SearchUserTaskAuditLogs
def search_user_task_audit_logs_example(user_task_key: UserTaskKey) -> None:
    client = CamundaClient()

    result = client.search_user_task_audit_logs(
        user_task_key=user_task_key,
        data=UserTaskAuditLogSearchQueryRequest(),
    )

    if not isinstance(result.items, Unset):
        for log in result.items:
            print(f"Audit log: {log.audit_log_key}")
# endregion SearchUserTaskAuditLogs


# region CreateDocument
def create_document_example() -> None:
    import io

    client = CamundaClient()

    result = client.create_document(
        data=CreateDocumentData(
            file=File(payload=io.BytesIO(b"hello world"), file_name="example.txt"),
        ),
    )

    print(f"Document ID: {result.document_id}")
# endregion CreateDocument


# region CreateDocuments
def create_documents_example() -> None:
    import io

    client = CamundaClient()

    result = client.create_documents(
        data=CreateDocumentsData(
            files=[
                File(payload=io.BytesIO(b"file one"), file_name="one.txt"),
                File(payload=io.BytesIO(b"file two"), file_name="two.txt"),
            ],
        ),
    )

    if not isinstance(result.created_documents, Unset):
        for doc in result.created_documents:
            print(f"Created document: {doc.document_id}")
# endregion CreateDocuments


# region GetDocument
def get_document_example(document_id: DocumentId) -> None:
    client = CamundaClient()

    result = client.get_document(document_id=document_id)

    print(f"File name: {result.file_name}")
# endregion GetDocument


# region SearchUserTaskEffectiveVariables
def search_user_task_effective_variables_example(user_task_key: UserTaskKey) -> None:
    client = CamundaClient()

    result = client.search_user_task_effective_variables(
        user_task_key=user_task_key,
    )

    if not isinstance(result.items, Unset):
        for var in result.items:
            print(f"Variable: {var.name}")
# endregion SearchUserTaskEffectiveVariables
